src/main/java/domains/task/use_case/add_task/AddTaskInteractor.java

```diff
@@ -0,0 +1,38 @@
+package domains.task.use_case.add_task;
+import app.task.AddTaskUseCaseFactory;
+import domains.permission.entity.Permission;
+import domains.task.entity.Task;
+
+public class AddTaskInteractor implements AddTaskInputBoundary{
+    private final AddTaskDataAccessInterface addTaskDataAccessObject;
+    private final AddTaskOutputBoundary addTaskPresenter;
+
+    public AddTaskInteractor(AddTaskDataAccessInterface addTaskDataAccessObject,
+                                AddTaskOutputBoundary addTaskOutputBoundary) {
+        this.addTaskDataAccessObject = addTaskDataAccessObject;
+        this.addTaskPresenter = addTaskOutputBoundary;
+    }
+
+
+    @Override
+    public void execute(AddTaskInputData addTaskInputData) {
+        String projectID = addTaskInputData.getProjectID();
+        String content = addTaskInputData.getContent();
+
+        if (projectID == null || content == null){
+            addTaskPresenter.prepareFailView("must have projectID and content");
+        }
+        else{
+            try{
+                Task task = new Task(null, projectID, addTaskInputData.getTaskName());
+
+                String taskID = this.addTaskDataAccessObject.addTask(projectID, task);
+                AddTaskOutputData addTaskOutputData = new AddTaskOutputData(taskID);
+                this.addTaskPresenter.printTaskID(addTaskOutputData);
```

Comment:
supposed to call SQL (TODO: Andrew)

Reply:
@lz2512 @laura1945 

I just found out that Todoist can filter by project ID
https://developer.todoist.com/rest/v2/#get-active-tasks

Thus, I think you can change `this.addTaskDataAccessObject.addTask` to return void, because the ID will be maintained by Todoist